src/io/import_off.cc

```diff
@@ -53,6 +53,17 @@ std::unique_ptr<PolySet> import_off(const std::string& filename, const Location&
     return true;
   };
 
+  auto getcolor = [&](const auto& word){
+    int c;
+    if (boost::contains(word, ".")) {
+      c = (int)(boost::lexical_cast<float>(word) * 255);
```

Comment:
The FAQ of [boost::lexical_cast()](https://www.boost.org/doc/libs/1_85_0/doc/html/boost_lexical_cast/frequently_asked_questions.html) mentions:
```
boost::lexical_cast has the behavior of std::stringstream, which uses num_get functions of std::locale to convert numbers.
```
which sounds like this would break with locales that do not use "." as a decimal separator?

Reply:
Looks like there's no way to disable that behavior and it is locale dependent:
```
#include <iostream>
#include <boost/lexical_cast.hpp>

int main(int argc, char** argv) {
    std::setlocale(LC_ALL, "");
    std::locale::global(std::locale(""));

    std::cout << "LC_ALL: " << std::setlocale(LC_ALL, NULL) << std::endl;
    std::cout << "LC_CTYPE: " << std::setlocale(LC_CTYPE, NULL) << std::endl;
    std::cout << "C++ locale: " << std::locale().name() << std::endl;

    float f = boost::lexical_cast<float>(argv[1]);
    printf("%f\n", f);
    return 0;
}
```
$ g++ -o x x.cc && LC_ALL=en_US.utf8 ./x 1.2
LC_ALL: en_US.utf8
LC_CTYPE: en_US.utf8
C++ locale: en_US.utf8
1.200000
```
```
g++ -o x x.cc && LC_ALL=de_DE.utf8 ./x 1.2
LC_ALL: de_DE.utf8
LC_CTYPE: de_DE.utf8
C++ locale: de_DE.utf8
terminate called after throwing an instance of 'boost::wrapexcept<boost::bad_lexical_cast>'
  what():  bad lexical cast: source type value could not be interpreted as target
Aborted
```
```
g++ -o x x.cc && LC_ALL=de_DE.utf8 ./x 1,2
LC_ALL: de_DE.utf8
LC_CTYPE: de_DE.utf8
C++ locale: de_DE.utf8
1,200000
```

It would be nice if we could stop messing with the global locale as some of the other imports.

Reply:
the `std::from_chars()`, available since c++17, parses numbers locale-independently, so sounds like a good choice here.

```c++
#include <iostream>
#include <charconv>

int main(int argc, char** argv) {
    std::setlocale(LC_ALL, "");
    std::locale::global(std::locale(""));

    std::cout << "LC_ALL: " << std::setlocale(LC_ALL, NULL) << std::endl;
    std::cout << "LC_CTYPE: " << std::setlocale(LC_CTYPE, NULL) << std::endl;
    std::cout << "C++ locale: " << std::locale().name() << std::endl;

    std::string_view num(argv[1]);
    float f;
    auto result = std::from_chars(num.data(), num.data() + num.length(), f);
    if (result.ec == std::errc{}) {
      printf("%f (left over '%s')\n", f, result.ptr);
    } else {
      printf("parse error\n");
    }
    return 0;
}
```

Parsing with `en_US` locale
```
g++ --std=c++17 x.cc && LC_ALL=en_US.utf8 ./a.out 1.2
LC_ALL: en_US.utf8
LC_CTYPE: en_US.utf8
C++ locale: en_US.utf8
1.200000 (left over '')
```

Parsing with locale that usually uses comma for decimal point (`de_DE`): Parses 1.2 correctly (and then here: prints in German locale)

```
g++ --std=c++17 x.cc && LC_ALL=de_DE.utf8 ./a.out 1.2
LC_ALL: de_DE.utf8
LC_CTYPE: de_DE.utf8
C++ locale: de_DE.utf8
1,200000 (left over '')
```

Parsing 1,2 would not work: it will parse the `1` and considers the `,2` as left-over. But that is exactly how we would like to see the parsing work:

```
g++ --std=c++17 x.cc && LC_ALL=de_DE.utf8 ./a.out 1,2
LC_ALL: de_DE.utf8
LC_CTYPE: de_DE.utf8
C++ locale: de_DE.utf8
1,000000 (left over ',2')
```

... and correctly complains about things not parseable at all:

```
g++ --std=c++17 x.cc && LC_ALL=en_US.utf8 ./a.out foobar
LC_ALL: en_US.utf8
LC_CTYPE: en_US.utf8
C++ locale: en_US.utf8
parse error
```

Also advantage: no exceptions.